### PR TITLE
add functionality to support auth from frontend

### PR DESCRIPTION
**Pull request by abhiandthetruth**

### CODE REVIEW

File: src/security_api.py [hunk 1]
```diff
@@ -29,8 +29,11 @@ def get_token_bearerAuth(credentials: HTTPAuthorizationCredentials = Depends(bea
     :rtype: TokenModel | None
     """
     try:
+        print("Found credentials", credentials.credentials)
```
[the remark this answers is not in the document]
Comment (abhiandthetruth) [reply src/security_api.py:32]: Yes until auth is ironed out. I'll want to keep the logs

File: src/apis/authentication_api.py [hunk 2]
```diff
@@ -48,6 +48,7 @@
             "content": None
         },
         401: {"description": "Unauthorized"},
+        404: {"description": "user not found"}
```
[the remark this answers is not in the document]
Comment (abhiandthetruth) [reply src/apis/authentication_api.py:51]: Krta hu

File: src/apis/items_api.py [hunk 3]
```diff
@@ -166,11 +166,12 @@ async def items_item_id_put(
...
     request: Request,
     item: Item = Body(None, description=""),
-    token: TokenModel = Depends(get_token_bearerAuth),
+    token: TokenModel =  Depends(get_token_bearerAuth),
 ) -> None:
     """Create a new wardrobe item."""
+    print("Found token", token, type(token))
     item = jsonable_encoder(item)
-    item.user_id = token.user_id
+    item['user_id'] = token.user_id
```
[the remark this answers is not in the document]
Comment (abhiandthetruth) [reply src/apis/items_api.py:174]: Don't know, it's easy to enforce through pre hooks and pretty configs. Does this block?

[the remark this answers is not in the document]
Comment (Shreya-7) [reply src/apis/items_api.py:174]: not blocking, have another ticket for both hooks and linter

File: src/apis/authentication_api.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (abhiandthetruth) [reply src/apis/authentication_api.py]: done

**Pull request merged**